mlflow/models/evaluation/base.py

```diff
@@ -957,24 +957,29 @@ def _get_last_failed_evaluator():
     return _last_failed_evaluator
 
 
+# BE CAREFUL WHEN EDITING SIGNATURE OF THIS FUNCTION.
+# This function is monkeypatched in Databricks codebase, and the order of arguments
+# must be kept the same, even if some of them are unused by MLflow.
```

Comment:
This might make legal or security folks unhappy...

Reply:
We can add another note: "Do not import this function into other modules, otherwise it will make Databricks patching losing effectiveness"

Reply:
We have another way to fix it, if you prefer, we can update Databricks side patching code to fit either new or old version `_evaluate`, and backport Databricks side fix to old runtime versions.

Reply:
> This might make legal or security folks unhappy...

Oh do you mean we shouldn't mention about patching in DB...?

> We have another way to fix it, if you prefer, we can update Databricks side patching code to fit either new or old version _evaluate, and backport Databricks side fix to old runtime versions.

Yes I'm doing that in parallel, but it takes several weeks until the change is fully deployed so I think we also need this band-aiding🙂

Reply:
> Oh do you mean we shouldn't mention about patching in DB...?

I think it's a gray zone. It leaks internal information. Can we just say "Do not change the argument order"?

Reply:
That's fair. The context wouldn't be lost in a few weeks/months until we fix the patching issue.